src/Storages/StorageMergeTree.cpp

```diff
@@ -1421,73 +1402,258 @@ MergeTreeDataPartPtr StorageMergeTree::outdatePart(MergeTreeTransaction * txn, c
 void StorageMergeTree::dropPartNoWaitNoThrow(const String & part_name)
 {
     if (auto part = outdatePart(NO_TRANSACTION_RAW, part_name, /*force=*/ false))
-        dropPartsImpl({part}, /*detach=*/ false);
+    {
+        if (deduplication_log)
+        {
+            deduplication_log->dropPart(part->info);
+        }
+
+        LOG_INFO(log, "Removed 1 parts.");
+
+        /// Need to destroy part objects before clearing them from filesystem.
+        part.reset();
+
+        clearOldPartsFromFilesystem();
+    }
 
     /// Else nothing to do, part was removed in some different way
 }
 
-void StorageMergeTree::dropPart(const String & part_name, bool detach, ContextPtr query_context)
+struct PartitionInfo
 {
-    if (auto part = outdatePart(query_context->getCurrentTransaction().get(), part_name, /*force=*/ true))
-        dropPartsImpl({part}, detach);
-}
+    MergeTreePartition partition{};
+    Int64 min_block = std::numeric_limits<Int64>::max();
+    Int64 max_block = 0;
+    UInt32 max_level = 0;
+    Int64 max_mutation = 0;
```

Comment:
What's the difference from `MergeTreePartInfo`?

Reply:
The structure PartitionInfo contains MergeTreePartition as partition. MergeTreePartInfo contains String as partition_id. For creating empty part MergeTreePartition have to be set. I decided not to constructing it from string or part name, but just copy from a part with the same partition.